### PR TITLE
PSD variation code

### PR DESCRIPTION
**Pull request by lauranuttall**
This pull request is for the PSD variation code I've been implementing along Alex's suggestions. 

This code calculates a PSD every 8 seconds with a 4 second overlap. It then calculates a 512 second PSD and divides the two, looking for variations between the longer and shorter PSD. Then by summing up the differences over a frequency range we care about (20-512Hz) we can gain a 'PSD variation statistic' which quantifies how different the PSDs are. This statistic value has been recorded for each trigger so that this value is saved at the output. The variable is called psd_var_val. 

I've tested this code with the usual pycbc_inspiral settings over a day's worth of data. Everything seems to be saved fine. I've also tested the code turning off these new settings and everything works fine. I'm currently just checking now how well this statistic picks up non-stationarity.

### CODE REVIEW

File: pycbc/events/events.py [hunk 1]
```diff
@@ -447,6 +447,8 @@ def __setitem__(self, name, data):
             f['bank_chisq_dof'] = self.events['bank_chisq_dof']
             f['cont_chisq'] = self.events['cont_chisq']
             f['end_time'] = self.events['time_index'] / float(self.opt.sample_rate) + self.opt.gps_start_time
+            #if self.opt.psdvar_short_segment is not None:
```
Comment (ahnitz) [inline pycbc/events/events.py]: Before merging, should probably remove the debug version.

File: pycbc/events/events.py [hunk 2]
```diff
@@ -672,6 +677,8 @@ def __setitem__(self, name, data):
                 f['end_time'] = ifo_events['time_index'] / \
                         float(self.opt.sample_rate[ifo_str]) + \
                         self.opt.gps_start_time[ifo_str]
+                #if self.opt.psdvar_short_segment is not None:
```
Comment (ahnitz) [inline pycbc/events/events.py]: another debug location to remove

File: pycbc/psd_variation.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
+""" PSD Variation """
```
Comment (ahnitz) [inline pycbc/psd_variation.py]: Maybe this file should be within the psd package? And simply renamed 'variation' so the namespace would be. 

pycbc.psd.variation

How does that sound?

**Comment by ahnitz** [comment]
Travis error is 
```

Warning, treated as error:
/home/travis/build/pycbc-build/environment/lib/python2.7/site-packages/PyCBC-113106-py2.7.egg/pycbc/psd_variation.py:docstring of pycbc.psd_variation.calc_psd_variation:6:Unexpected indentation.
make: *** [html] Error 2
Traceback (most recent call last):
  File "setup.py", line 510, in <module>
    'pycbc.tmpltbank': find_package_data('pycbc/tmpltbank')},
  File "/home/travis/build/pycbc-build/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/home/travis/build/pycbc-build/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/home/travis/build/pycbc-build/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "setup.py", line 331, in run
    stderr=subprocess.STDOUT, shell=True)
  File "/home/travis/build/pycbc-build/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'mkdir -p _gh-pages/latest && touch _gh-pages/.nojekyll && cd docs; cp Makefile.gh_pages Makefile; cp conf_std.py conf.py; sphinx-apidoc  -o ./ -f -A 'PyCBC dev team' -V '0.1' ../pycbc && make html' returned non-zero exit status 2
---------------------------------------------------------
```

File: pycbc/psd_variation.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+from pycbc.types import TimeSeries, zeros
+
+def calc_psd_variation(strain, psd_short_segment, psd_long_segment, 
+                       overlap, low_freq, high_freq):
+    """Calculates time series of PSD variability
+
+    This function first splits the segment up in to 512 second chunks. It 
+    then calculates the PSD over this 512 second period as well as in 4 
+    second chunks throughout each 512 second period. Next the function 
+     estimates how different the 4 second PSD is to the 512 second PSD and
```
Comment (ahnitz) [inline pycbc/psd_variation.py]: I think travis is complaining about the indentation starting here.

File: pycbc/events/events.py (hunk 1 above)
Comment (lauranuttall) [reply pycbc/events/events.py]: yep, removed

File: pycbc/events/events.py (hunk 2 above)
Comment (lauranuttall) [reply pycbc/events/events.py]: done

File: pycbc/psd_variation.py (hunk 3 above)
Comment (lauranuttall) [reply pycbc/psd_variation.py]: sounds good to me, moved

File: pycbc/psd_variation.py (hunk 4 above)
Comment (lauranuttall) [reply pycbc/psd_variation.py]: good catch, have corrected

File: pycbc/psd/variation.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+from pycbc.types import TimeSeries, zeros
+
+def calc_psd_variation(strain, psd_short_segment, psd_long_segment, 
+                       overlap, low_freq, high_freq):
+    """Calculates time series of PSD variability
+
+    This function first splits the segment up in to 512 second chunks. It 
+    then calculates the PSD over this 512 second period as well as in 4 
+    second chunks throughout each 512 second period. Next the function 
+     estimates how different the 4 second PSD is to the 512 second PSD and
```
Comment (ahnitz) [inline pycbc/psd/variation.py]: Spacing still looks off, which is what I think the documentation build is complaining about.

**Comment by ahnitz** [comment]
@lauranuttall Can you add the email you use for commits to github as well so you get credit for your contributions?

**Comment by ahnitz** [comment]
@lauranuttall The unused import should probably be fixed, but otherwise, this PR is approved. Please merge whenever you like.

**Comment by ahnitz** [comment]
https://app.codacy.com/app/thomas-dent/pycbc/pullRequest?prid=1350411

**Pull request merged**